.github/workflows/build.yml

```diff
@@ -98,3 +98,38 @@ jobs:
       - name: Setup tmate debug session
         uses: mxschmitt/action-tmate@v3
         if: env.RUN_TMATE
+  test:
+    name: test source
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Download Go
+        uses: actions/setup-go@v4
+        with:
+          go-version: '^1.20.2'
+        id: go
+
+      - name: Check out code into the Go module directory
+        uses: actions/checkout@v3
+
+      - name: Retrieve package location
+        run: |
+            echo PACKAGE_LOCATION="$(find ./ -name 'test')/../" >> $GITHUB_OUTPUT
+        id: package-location
+
+      - name: Generate mocks and run tests
+        working-directory: ${{ steps.package-location.outputs.PACKAGE_LOCATION }}
+        run: |
+          export GOPATH="$HOME/go/"
+          export PATH=$PATH:$GOPATH/bin
+          go install github.com/golang/mock/mockgen@latest
+          go install github.com/jstemmer/go-junit-report@latest
+          export PACKAGE_LOCATION="$(find ./ -name 'test')/../"
+          go generate ./
+          go test -v ./test/* | go-junit-report -set-exit-code > report.xml
+
+      - name: Test summary
+        uses: test-summary/action@v1
+        with:
+          paths: |
+            ${{ steps.package-location.outputs.PACKAGE_LOCATION }}/report.xml
+        if: always()
```

Comment:
Can you please add a brief comment here explaining why this step should always be executed, even when the job is canceled?

Reply:
Okay I will remove it. I didn't account for cancellation. The idea was if previous steps fail we would still want to update the build status since it will be affected.

Reply:
Looks like https://github.com/cisagov/skeleton-golang-package/pull/2/commits/f57fab7e614fac126f2cf8b3828b69e7ccb147af was clobbered by a subsequent force push.  @jasonodoom Can you please fix that?

Reply:
My bad. Please see commit https://github.com/cisagov/skeleton-golang-package/commit/2637241db6f52824bf2007c8f8529a98ad8192ab